controllers/machine_controller_phases.go

```diff
@@ -114,6 +114,17 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, cluster *clus
 		return external.ReconcileOutput{}, err
 	}
 
+	// remove machineset controller onwer, if any since the Machine should be the owner.
+	if controller := metav1.GetControllerOf(obj); controller != nil {
```

Comment:
minor nit, but maybe check if controller.Kind == "MachineSet" ?

Reply:
Doesn't the hardcoded value for ownerRef.Kind to `MachineSet` implicitly do the same thing?

Reply:
Yeah, it does. If I understand correctly, for performance, without checking Kind it'll always go through all ownerreferences, no matter if the MacineSet is present or not.  Also, for readability, this might be a bit confusing, as we kinds of assume this controller is MachineSet, but in fact it maybe Machine.

Reply:
I still feel adding the check for controller Kind, somehow duplicates the logic. I will defer to the better judgement of experienced folks like @vincepri

Reply:
My two cents.
I'm +1 to have the check on "controlled by a machineSet" more explicit for improving for readability.

For the same reason, might be worth to split the command in two-line so we will have `ownerRefs := util.RemoveOwnerRef(...)` which makes it clear we are removing from the list, and the `obj.SetOwnerReferences(ownerRefs)` in a separated line

Reply:
It might be worth adding it for readability, + a comment explaining the reason why this has to be removed this way. In general, owner reference is a very finite set, performance-wise we wouldn't get many improvements.